runtime/criusupport/criusupport.cpp

```diff
@@ -890,7 +890,13 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		Trc_CRIU_checkpoint_nano_times(currentThread, checkpointNanoTimeMonotonic, checkpointNanoUTCTime);
 		TRIGGER_J9HOOK_VM_PREPARING_FOR_CHECKPOINT(vm->hookInterface, currentThread);
 
+		/* GC releases threads in a multi-thread fashion. Threads will need to remove
+		 * themselves from the threadgroup which requires a lock. Disable deadlock detection
+		 * temporarily while this happens.
+		 */
+		vm->extendedRuntimeFlags2 &= ~J9_EXTENDED_RUNTIME2_CRIU_SINGLE_THREAD_MODE;
 		vm->memoryManagerFunctions->j9gc_prepare_for_checkpoint(currentThread);
+		vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_CRIU_SINGLE_THREAD_MODE;
```

Comment:
While running `j9gc_prepare_for_checkpoint(currentThread)`, the non-checkpoint thread(s) such as `wait/sleep/park` (`threadhelp.cpp:timeCompensationHelper`) in single-threaded mode has a small window to proceed and will stop eventually due to `J9_PUBLIC_FLAGS_HALT_THREAD_FOR_CHECKPOINT` set for those threads.
There is no risk of `JVMCheckpointException` since the hooks are already finished.
The only side-effect I see is that this short run might affect the application hook results if it assumes all non-checkpoint threads are paused, do we accept such a small restriction?

Reply:
Ya, I was thinking about this, perhaps we need two flags